src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -309,24 +309,37 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 
 	private getCurrentSelectionText(): string {
 		if (this.cellModel.currentMode === CellEditModes.WYSIWYG) {
-			return document.getSelection()?.toString() || '';
+			return document.getSelection()?.anchorNode.nodeValue || '';
 		} else {
 			const editorControl = this.getCellEditorControl();
 			const selection = editorControl?.getSelection();
 			if (selection && !selection.isEmpty()) {
 				const textModel = editorControl?.getModel() as TextModel;
 				const value = textModel?.getValueInRange(selection);
-				return value || '';
+				let linkLabel = value.substring(value.indexOf('[') + 1, value.indexOf(']'));
```

Comment:
Probably not super common but won't this logic break if the text included brackets or parentheses?

Reply:
Only break on the ']', since the first one would get the appropriate starting bracket but we would get the first index of the closing bracket which could possibly be included in the text. 
I can try and ensure that the last bracket is the last one found in the string so that even if brackets are included in the title they would also appear. I don't believe we would have to worry about parentheses since links should not contain them as far as I know.

Reply:
But if the text contains parentheses then it would still be an issue. For example getCurrentLinkUrl() of `[test()](url.com)` would return an empty string instead of url.com

Reply:
ah so the last index of the both the '(' and ')' would be needed as well.